### PR TITLE
Don't pass OPENSHIFT_IP env variable when building old BWC branches

### LINKED ISSUE #50025: Multiple buildOSS failures for Windows and MacOS
Multiple builds have failed recently and all seemingly due to the same cause.

All the windows OSS builds failed for 6.8.6 due to `Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal()`.

MacOS failed on 6.8.6 for the same.

Windows OSS Builds failed on 7.5.1 due to ` Failed to ping owner of lock for artifact cache (C:\Users\jenkins.PACKER-5DA3F31F\.gradle\caches\modules-2)`

Admittedly, the 7.5.1 and the 6.8.6 build failures might be unrelated. 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-windows-compatibility/os=windows-2012-r2/305/console

```
07:25:24  [6.8.6] FAILURE: Build failed with an exception.
07:25:24  [6.8.6] 
07:25:24  [6.8.6] * What went wrong:
07:25:24  [6.8.6] Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
07:25:24  [6.8.6] > Failed to ping owner of lock for journal cache (C:\Users\jenkins.PACKER-5DA3F31F\.gradle\caches\journal-1) (lock id: 1364286758791752371, port: 53887)
...
07:25:26  [7.5.1] * What went wrong:
07:25:26  [7.5.1] Could not resolve all dependencies for configuration 'classpath'.
07:25:26  [7.5.1] > Failed to ping owner of lock for artifact cache (C:\Users\jenkins.PACKER-5DA3F31F\.gradle\caches\modules-2) (lock id: 4060380626467574440, port: 53887)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-darwin-compatibility/384/console

```
07:34:40  [6.8.6] FAILURE: Build failed with an exception.
07:34:40  [6.8.6] 
07:34:40  [6.8.6] * What went wrong:
07:34:40  [6.8.6] Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
07:34:40  [6.8.6] > Failed to ping owner of lock for journal cache (/private/var/lib/jenkins/.gradle/caches/journal-1) (lock id: -5519563589571878859, port: 64009)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-windows-compatibility/os=windows-2016/305/console

```
07:35:39  [6.8.6] FAILURE: Build failed with an exception.
07:35:39  [6.8.6] 
07:35:39  [6.8.6] * What went wrong:
07:35:39  [6.8.6] Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
07:35:39  [6.8.6] > Failed to ping owner of lock for journal cache (C:\Users\jenkins.PACKER-5DA3F31E\.gradle\caches\journal-1) (lock id: -6119171925434326694, port: 57853)
...
07:35:41  [7.5.1] * What went wrong:
07:35:41  [7.5.1] Could not resolve all dependencies for configuration 'classpath'.
07:35:41  [7.5.1] > Failed to ping owner of lock for artifact cache (C:\Users\jenkins.PACKER-5DA3F31E\.gradle\caches\modules-2) (lock id: -6949790552055040817, port: 57853)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-windows-compatibility/os=windows-2019/305/console
```
07:43:40  [6.8.6] * What went wrong:
07:43:40  [6.8.6] Could not create service of type FileAccessTimeJournal using GradleUserHomeScopeServices.createFileAccessTimeJournal().
07:43:40  [6.8.6] > Failed to ping owner of lock for journal cache (C:\Users\jenkins.PACKER-5DA3F2D9\.gradle\caches\journal-1) (lock id: -4306243589339416205, port: 61491)
...
07:43:43  [7.5.1] * What went wrong:
07:43:43  [7.5.1] Could not resolve all dependencies for configuration 'classpath'.
07:43:43  [7.5.1] > Failed to ping owner of lock for artifact cache (C:\Users\jenkins.PACKER-5DA3F2D9\.gradle\caches\modules-2) (lock id: 4398689666861551841, port: 61491)
```

### PR DESCRIPTION
**Pull request by mark-vieira**
This commit tweaks the workaround introduced in #49211 to support
Gradle 6.0. In the workaround, we specifically override the address
the Gradle daemon binds to by passing the desired address via the
OPENSHIFT_IP environment variable. This works fine for builds using
Gradle 6.0, but for older Gradle versions this causes issues with
inter-daemon communication, specifically when we build BWC branches
not on Gradle 6.0. The fix here is to strip that environment variable
out when building the target BWC branch if that branch is on an
older Gradle version.

This is all temporary and will be removed when this bug fix is released
in Gradle 6.1.

Closes #50025

### CODE REVIEW

File: distribution/bwc/build.gradle [hunk 1]
```diff
@@ -161,6 +162,13 @@ bwcVersions.forPreviousUnreleased { BwcVersions.UnreleasedVersionInfo unreleased
         spoolOutput = true
         workingDir = checkoutDir
         doFirst {
+          // TODO: Remove this once we've updated to Gradle 6.1
+          // Workaround for https://github.com/gradle/gradle/issues/11426
+          if (GradleVersion.version(file("${ checkoutDir}/buildSrc/src/main/resources/minimumGradleVersion").text) != GradleVersion.current()) {
+            environment = environment.findAll { key, val -> key != 'OPENSHIFT_IP' }
+            environment.forEach { key, val -> println("$key=$val") }
```
Comment (rjernst) [inline distribution/bwc/build.gradle]: Is this leftover debug?

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Review by rjernst — APPROVED** [review]
LGTM

File: distribution/bwc/build.gradle (hunk 1 above)
Comment (mark-vieira) [reply distribution/bwc/build.gradle]: Yes, I'll remove this once everything is green.

**Comment by mark-vieira** [comment]
@elasticmachine retest this please

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**